### PR TITLE
Travis CI:  Use services: xvfb and $TRAVIS_PYTHON_VERSION

### PR DESCRIPTION
**Pull request by cclauss**
https://docs.travis-ci.com/user/gui-and-headless-browsers/#using-services
https://docs.travis-ci.com/user/environment-variables

- [ ] closes #xxxx
- [x] tests added / passed. 
- [ ] code/documentation is well formatted.  
- [ ] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in master and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/master -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ jobs:
         - LINT=true
     - python: 3.9
       env:
-        - PYTHON=3.7
```
Comment (bashtage) [inline .travis.yml:52]: Let's move the doc build to 3.8.  It has a long tail of deps and better for it to be stable.

**Comment by bashtage** [comment]
I think we need to wait until SciPy has wheels for 3.9 before merging this, and then to move some of the allowed failures to required.

**Comment by bashtage** [comment]
I also think pip-pre should stay 3.8 for now since it is probably the Python that most are still using.

**Comment by bashtage** [comment]
LGTM.  I guess 3.9 isn't ready yet.

**Comment by cclauss** [comment]
Python 3.9 is clearly ready but statsmodels still runs Python 3.8 in `allow_failures` mode.

**Comment by bashtage** [comment]
The allowed failure is for pre release packages.  It helps pick up changes early.  3.9 should be a required pass.

**Pull request merged**